### ISSUE TITLE
So, ah, how do you start it?

### ISSUE THREAD

**Issue by lolsatdanger** [rezreal/gentlemans-gallery#2]
The configuration page displays, but nothing happens after uploading an image folder. There doesn't seem to be a way to start the gallery.

Is this intended to be downloaded and run locally? Any recommendations on how best to locally build it if so?

Unsure if it isn't working on my end, or just isn't working.

**Comment by rezreal** [comment]
normally after adding a folder a button should appear at the bottom of the page, saying "Start here (nn slides)!" (admitted: not a nice ux).

**Comment by rezreal** [comment]
As far as I remember it must run on a https origin or localhost, otherwise certain features will not work. You should be able to build and run it with `npm install && npm run start`

**Comment by lolsatdanger** [comment]
I got it running on http://localhost:4200/ doing npm install, tweaking the `nx serve --host 0.0.0.0 --ssl true` in the package to just `nx serve` or throwing my local ip in replacing the 0.0.0.0 and then doing `npm run start`. Unfortunately, it seems to behave the same as on https://rezreal.github.io/gentlemans-gallery/ -- the button doesn't pop after selecting a folder, or after selecting the sample images. It doesn't seem to just be a display issue, as I also can't find it if I inspect the html in my browser and look for the button (I assume it is the one at line 734 in MainComponent.tsx based on what you said?) 

I did get an error message when doing the npm run start of  `Browserslist: caniuse-lite is outdated. Please run the following command: npm update` which looks to be gone now with `npx browserslist@latest --update-db` but that didn't resolve the button not appearing (not that I would've expected it to).


I get this output in the console if I hit the load sample images button: ![load sample images output](https://user-images.githubusercontent.com/55366558/140837566-f8ce473c-999b-4c67-b992-ecb0518ad3d6.PNG)

So it appears to do something, but the button doesn't pop. I've tried in a couple different browsers as well.

If I forcibly make the button load by removing the criteria to show it or not, it always displays as 0 and upon clicking the button after loading the sample images I get a page that says "You made it! Your score: 0. Hit reload to start over." (There is no reload button shown.)

I know this is a relatively old project of yours, but it's an extremely interesting one to me and I'd appreciate any help if you have any ideas about what the issue could be 🙂 

Edit: I think it may have something to do with the slides array? It doesn't seem to populate when images are loaded looking at the output in the console, but the imageFiles array did.

**Comment by rezreal** [comment]
I agree, load demo images does not seem to work anymore :) But it still works for me with some of my example folders.

How is your folder structured? did you follow the instructions (providing a folder with both, the images and purify json files)? is the folder "flat" (has no subfolders)?

The button should appear at the very end of the page

**Comment by rezreal** [comment]
could you provide a sample json? maybe the structure output of purify has changed in an incompatible way

**Comment by rezreal** [comment]
TBH I need to take some time to  bring the feature detection into the browser so no offline preprocessing would be required :)

**Comment by lolsatdanger** [comment]
Looked into it a lot more yesterday and made some tweaks https://github.com/lolsatdanger/gentlemans-gallery

The button just wasn't popping for me, but always displaying it seems like a good workaround anyway. I may have been too focused on the sample button because yes, this seems like it was just not working with the load sample image button and works fine with actual image folders you make.

Then other issues I had, cayote didn't work. I snagged the code you used in https://github.com/rezreal/coyote and stuffed it in to replace the old code and now it connects to my dg-labs as expected. (I sometimes have the eStim lagging a bit behind though?)

And then also I personally feel censoring the parts you are not supposed to look at somewhat defeats the purpose so I nixed that and the boxes. Would be neat to make this into a setting. (To me it doesn't make sense to remove the temptation entirely by also censoring the images, I just want it to be able to know where I can and cannot look)


Besides these fixes, alternate gallery mode doesn't seem to work right now, but this version otherwise appears functional doing npm run start. Really glad you had mouse support too, so that cheap people like me without a fancy piece of eye tracking hardware can use a webcam mouse eye tracker setup instead haha.

Would be nice if images could resize to some standard, and they also seem to overlap the top bar, but those seem like relatively easy layout tweaks that I'll experiment with at some point if you do not haha.

This is awesome work, thank you!
